Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -1921,6 +1924,34 @@ repoTypeDirname Bazaar    = [".bzr"]
 repoTypeDirname Monotone  = ["_MTN"]
 repoTypeDirname Pijul     = [".pijul"]
 
+checkDesirableExtraSrcFilesAreIncluded :: Monad m => CheckPackageContentOps m
+                                       -> PackageDescription
+                                       -> m [PackageCheck]
+checkDesirableExtraSrcFilesAreIncluded ops pkg = do
+  let dir = "."
+  rootContents <- getDirectoryContents ops dir
+  desirable <- filterM (doesFileExist ops)
+    [file | file <- rootContents, isDesirableExtraSrcFile file]
+
+  let included = map normalise $ extraSrcFiles pkg
+  let globs = rights $ map (parseFileGlob (specVersion pkg)) included
+  let unlisted = filter (\fp ->
+                          not $ any (\g -> isJust $ fileGlobMatches g fp) globs
+                        ) (desirable \\ included)
+
+  return [ PackageDistSuspiciousWarn $
```

Comment:
> Warning: These warnings may cause trouble when distributing the package:

Is not correct.

There needs to be even weaker level. 

---

I have packages where README is just for GitHub to not look stupid. All "README" is already in main module documentation or package description.

(`ghc-options: -O2` having `PackageDistSuspiciousWarn` is acceptable, it's a fluke we ought to fix: If `cabal-install` is compiling with `optimizations: False` it would be great if packages weren't `-O2`, but if optimizations are enabled package authors could choose prefer `-O2` over `-O1` (or some other additional GHC flags). I.e. `.cabal` format isn't descriptive enough. Missing README isn't that level of trouble.)